### PR TITLE
Support datetime serialization in JSON responses

### PR DESCRIPTION
**Pull request by kashizui**
Fixes codalab/codalab-worksheets#235 (stats on worksheets.codalab.org doesn't load)

Also ensure that future JSON serialization errors send us emails.

### CODE REVIEW

**Comment by kashizui** [comment]
@percyliang I don't really like my current solution, where do you think would be the best place to make sure that 'frozen' is properly serialized?

File: codalab/rest/legacy.py [hunk 1]
```diff
@@ -145,6 +145,16 @@ def search_worksheets(keywords, worksheet_uuid=None):
...
+        # Hack: Make sure worksheet infos are JSON serializable.
+        # Since this method is only used for worksheet interpretation (i.e.
+        # resolving wsearch directives), these worksheet infos are never
+        # serialized properly through WorksheetSchema or the Worksheet
+        # ORMObject, and so the `frozen` attribute will cause a serialization
+        # error when we attempt to serialize the interpretation results into
+        # JSON.
+        for info in results:
+            if isinstance(info['frozen'], datetime):
+                info['frozen'] = str(info['frozen'])
```
Comment (percyliang) [inline codalab/rest/legacy.py]: I wouldn't use str()...use isoformat().

I think the `BundleModel` should be responsible for returning things back which are JSON serializable.  This method should not be looking at the internals of `results`.

**Review by percyliang — CHANGES_REQUESTED** [review]

**Comment by kashizui** [comment]
+1 on isoformat().

I generally agree, but BundleModel.search_worksheets() is also used by code
that depends on datetime fields being actual datetime objects, so I had
avoided modifying the model method.

**Comment by percyliang** [comment]
We could just convert the isoformat string back to a datetime when we need it, although that's a bit backwards...

Alternatively, we can have a general utility function that converts any JSON object with datetimes (and perhaps other non-serializable types) into a serializable JSON?

**Comment by kashizui** [comment]
@percyliang Pushed new changes that handles all datetime objects in JSON response objects. The `default` function handles any nonstandard objects that the JSON encoder encounters and cannot serialize.

**Comment by kashizui** [comment]
@percyliang reminder to review again

**Review by percyliang — APPROVED** [review]
Nice LGTM

**Pull request merged**